Gdip_All.ahk

```diff
@@ -2812,7 +2812,7 @@ Gdip_PixelateBitmap(pBitmap, &pBitmapOut, BlockSize)
 		nCount := StrLen(MCode_PixelateBitmap)//2
 		loop nCount
 			NumPut("UChar", "0x" SubStr(MCode_PixelateBitmap, (2*A_Index)-1, 2), PixelateBitmap, A_Index-1)
-		DllCall("VirtualProtect", "UPtr", PixelateBitmap.Ptr, "UPtr", PixelateBitmap, "UInt", 0x40, "UPtr*", 0)
```

Comment:
according to here, the third parameter should be the size of the target region
https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualprotect

I still have no idea what line 2831 does (what does it mean to make a dllcall with a buffer as the call name?), but the example definitely looks functional with an animated varying pixelation effect

Reply:
The 2831 call uses the option to use a pointer to a function to call instead of name. Look at https://www.autohotkey.com/docs/v2/lib/DllCall.htm: "This parameter may also consist solely of an integer, which is interpreted as the address of the function to call. Sources of such addresses include [COM](https://www.autohotkey.com/docs/v2/lib/DllCall.htm#COM) and [CallbackCreate](https://www.autohotkey.com/docs/v2/lib/CallbackCreate.htm)."
So in other words this one calls the MCode that you can see in the MCode_PixelateBitmap assignment.

Reply:
oh that's wild, I didn't realize those chunks were actually code. Makes sense now, thanks!